### PR TITLE
Changing gwd_opt to a domain-dependent namelist

### PR DESCRIPTION
**Pull request by weiwangncar**
TYPE: enhancement

KEYWORDS: GWD, domain dependent

SOURCE: internal

DESCRIPTION OF CHANGES: 
To assist the possibility that a user may turn off GWD in a nest, this PR make gwd_opt domain-dependent.

LIST OF MODIFIED FILES: 
M       Registry/Registry.EM_COMMON
M       share/module_check_a_mundo.F
M       run/README.namelist
M       test/em_real/namelist.input
M       test/em_real/namelist.input.global
M       test/em_real/namelist.input.jan00
M       test/em_real/namelist.input.jun01

TESTS CONDUCTED: 
Code tested with gwd_opt set to 1,1 and 1,0. The code behaves as expected. It passed automatic jenkins test.

RELEASE NOTE: 
The gwd_opt namelist has been changed to domain-dependent namelist. Hence one may turn off the option in a high-resolution domain.

### CODE REVIEW

File: share/module_check_a_mundo.F [hunk 1]
```diff
@@ -1425,6 +1425,28 @@ END FUNCTION bep_bem_nbui_max
             count_fatal_error = count_fatal_error + 1
          END IF
 
+!-----------------------------------------------------------------------
```
Comment (davegill) [inline share/module_check_a_mundo.F]: This is accidentally in here, right? We are focussing on GWD.

**Comment by davegill** [comment]
@weiwangncar 
Wei,
I checked the following:
1. The only location in check_a_mundo for GWD is this new one. OK
2. There are a couple of locations in NMM that make some assumptions about GWD and the
domain id:
```
dyn_nmm/module_PHYSICS_CALLS.F:      IF (grid%gwd_opt .eq. 2 .AND. grid%id.eq.1) THEN 
```
```
dyn_nmm/start_domain_nmm.F:   IF(grid%gwd_opt .eq. 2 .AND. grid%id .eq. 1 .AND. allowed_to_read) THEN
```
3. I know that 4dvar only does one domain at a time. Is this OK?
```
wrftladj/module_adtl_grid_utilities.F:   CALL nl_get_gwd_opt( 1, gwd_opt )
```

**Comment by weiwangncar** [comment]
@davegill gwd_opt is already a domain-dependent namelist in NMM. So I think that should be ok. The jenkins test can tell. I'm not sure about 4dvar. Do we have a test using jenkins?

**Comment by weiwangncar** [comment]
@davegill This PR has passed jenkins test.

**Pull request merged**